app/src/main/java/ch/epfl/sweng/SDP/game/drawing/DrawingOnline.java

```diff
@@ -0,0 +1,166 @@
+        if (!isVotingActivityLaunched) {
+            Matchmaker.getInstance(Account.getInstance(this)).leaveRoom(roomId);
+        }
+
+        removeAllListeners();
+
+        finish();
+    }
+
+    protected void removeAllListeners() {
+        timerRef.removeEventListener(listenerTimer);
+        stateRef.removeEventListener(listenerState);
+    }
+
+    protected final ValueEventListener listenerState = new ValueEventListener() {
+        @Override
+        public void onDataChange(@NonNull DataSnapshot dataSnapshot) {
+            Integer state = dataSnapshot.getValue(Integer.class);
+            if(state != null) {
+                GameStates stateEnum = GameStates.convertValueIntoState(state);
+                switch(stateEnum) {
+                    case START_VOTING_ACTIVITY:
+                        isVotingActivityLaunched = true;
+                        timerRef.removeEventListener(listenerTimer);
+                        Intent intent = new Intent(getApplicationContext(),
+                                VotingPageActivity.class);
+                        Log.d(TAG,winningWord);
+                        intent.putExtra("RoomID", roomId);
+                        startActivity(intent);
+                        break;
+                    default:
+                }
+            }
+        }
+
+        @Override
+        public void onCancelled(@NonNull DatabaseError databaseError) {
+            // Does nothing for the moment
+        }
+    };
+
+
+    @Override
+    public boolean onKeyDown(int keyCode, KeyEvent event) {
+        if ((keyCode == KeyEvent.KEYCODE_BACK)) {
+            Matchmaker.getInstance(Account.getInstance(this))
+                    .leaveRoom(roomId);
+            launchActivity(HomeActivity.class);
+            finish();
+        }
+        return super.onKeyDown(keyCode, event);
+
+    }
+
+    // MARK: COUNTDOWN METHODS
+
+    /**
+     * Gets called when time is over.
+     * Saves drawing in database and storage and calls new activity.
+     */
+    private void stop() {
+        LocalDbHandlerForImages localDbHandler = new LocalDbHandlerForImages(this, null, 1);
+        paintView.saveCanvasInDb(localDbHandler);
+        paintView.saveCanvasInStorage();
+        // add redirection here
+    }
+
+    /**
+     * Initializes the countdown to a given time.
+     *
+     * @return the countdown
+     */
+    private CountDownTimer setCountdownTimer() {
+        return new CountDownTimer(time, timeInterval) {
+            public void onTick(long millisUntilFinished) {
+                TextView textView = findViewById(R.id.timeRemaining);
+                textView.setText(Long.toString(millisUntilFinished / timeInterval));
+            }
+
+            public void onFinish() {
+                TextView textView = findViewById(R.id.timeRemaining);
+                textView.setTextSize(20);
+                textView.setText("Time over!");
+                stop();
+            }
+        }.start();
+    }
+
```

Comment:
This is a local timer thus in online mode is no longer useful since we are taking the timer form firebase.